pages/home.jsx

```diff
@@ -146,11 +149,121 @@ var BlogSection = React.createClass({
+      e.preventDefault();
+      alert("PLEDGE_MAILINGLIST_URL is not defined. Simulating " +
+            "a successful pledge signup now.");
+      window.location = "?pledge=thanks";
+    }
+  },
+  renderValidationErrors: function() {
+    if (this.state.validationErrors.length) {
+      return (
+        <div className="alert alert-danger">
+          <p className="error-msg">Please enter an email address.</p>
+        </div>
+      );
+    }
+  },
+  render: function() {
+    return (
+      <Modal modalTitle="" foldedStyle className="modal-pledge">
+        <ImageTag className="image center-block"
+                  src1x="/img/pages/home/svg/icon-teach-man-chalkboard-pledge.svg"
+                  alt="" width={150} height={150} />
+        <p>Because the Web is a global public resource that's integral to modern life, <strong>I pledge to:</strong></p>
+        <ul>
+          <li>teach others digital literacy skills through hands-on making</li>
+          <li>help people move beyond simply consuming the web to creating it</li>
+          <li>be a passionate advocate of an open and accessible web</li>
+        </ul>
+        <p><strong>Pledge now, and we'll follow up with with details about how you can teach the web!</strong></p>
+        <form className="mailinglist-signup" action={process.env.PLEDGE_MAILINGLIST_URL} method="POST" onSubmit={this.handleSubmit}>
+          <fieldset>
+            <label htmlFor="pledge-email" className="sr-only">email</label>
+            <div className="icon-field-container">
+              <i className="fa fa-envelope"></i>
+              <input id="pledge-email" name="email" type="email" size="30" placeholder="email@example.com" valueLink={this.linkState("email")} required />
+            </div>
+            <label htmlFor="pledge-pp-note" className="sr-only">I'm okay with you handling this info as you explain in your <a href="https://www.mozilla.org/en-US/privacy/websites/">privacy policy</a></label>
+            <input id="pledge-pp-note" name={process.env.PLEDGE_MAILINGLIST_PRIVACY_NAME} type="checkbox" checked readOnly required hidden />
+            <p className="pp-note">&#10003; I'm okay with you handling this info as you explain in your <a href="https://www.mozilla.org/en-US/privacy/websites/">privacy policy</a>.</p>
+            {this.renderValidationErrors()}
+          </fieldset>
+          <input type="submit" value="Submit Email" className="btn btn-awsm center-block" />
+        </form>
+      </Modal>
+    )
+  }
+});
+
+var ThankYouModal = React.createClass({
+  render: function() {
+    return (
+      <Modal modalTitle="Thanks for your pledge!" className="modal-pledge">
+        <p>We appreciate your commitment to keeping the web open, accessible and ours.</p>
+        <p><strong>Share and tell your friends</strong></p>
+        <div className="social-share">
+          <a href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fteach.mozilla.org" className="facebook">
```

Comment:
I like the idea of using `encodeURIComponent`! :+1: However I think it makes sense to hard code the URL "https://teach.mozilla.org" here instead of using `require('../lib/config').ORIGIN` as it might be confusing to people who are QAing on the staging site.

How about passing this `twitterShare` var  to `href` to make the code more readable?
```javascript
var twitterShare = "https://twitter.com/home?status=" +
                   encodeURIComponent("I just pledged to #TeachTheWeb! https://teach.mozilla.org");
```